### PR TITLE
Created a basic update profile image

### PR DESCRIPTION
**Pull request by dhanzy**
### What this PR does (required):
- Allows update of change profile image

### Screenshots / Videos (required):
- 
Screenshot (29)

### Any information needed to test this feature (required):
- Login and click on the edit profile on the dashboard. You will be directed to the profile page where you can upload your profile picture.

### Any issues with the current functionality (optional):
- Key errors on console due to unknown list

### CODE REVIEW

File: client/src/components/AvatarDisplay/AvatarDisplay.tsx [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-  user: User;
   style?: React.CSSProperties | undefined;
 }
 
-const AvatarDisplay = ({ user, style }: Props): JSX.Element => {
-  return <Avatar style={style} alt="Profile Image" src={`https://robohash.org/${user.email}.png`} />;
+const AvatarDisplay = ({ style }: Props): JSX.Element => {
+  const { loggedInUser } = useAuth();
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
```
Comment (rajivtitus) [inline client/src/components/AvatarDisplay/AvatarDisplay.tsx]: Maybe just perform one check here as opposed to three if clauses?

File: client/src/interface/User.ts [hunk 2]
```diff
@@ -1,8 +1,11 @@
+import { Profile } from './Profile';
+
 export interface User {
   email: string;
   username: string;
   _id: string;
   id: string;
```
Comment (rajivtitus) [inline client/src/interface/User.ts:7]: Why do we have two id's in our User interface? Please make changes if possible :)

File: client/src/pages/Profile/Profile.tsx [hunk 3]
```diff
@@ -35,11 +45,64 @@ function Profile(): JSX.Element {
...
+  const [previewUrl, setPreviewUrl] = useState<string>();
+  const [processing, setProcessing] = useState<boolean>(false);
+  const { loggedInUser } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    return <CircularProgress />;
+  }
+  if (loggedInUser.profile === undefined) return <CircularProgress />;
```
Comment (rajivtitus) [inline client/src/pages/Profile/Profile.tsx]: Same as comment above for performing a check with if clauses

File: client/src/pages/Profile/Profile.tsx [hunk 4]
```diff
@@ -35,11 +45,64 @@ function Profile(): JSX.Element {
...
+
+  const imageUpload = () => {
+    if (currentFile) {
+      setProcessing(true);
+      uploadImage({ image: currentFile }).then((resp) => {
+        if (resp.success) {
+          updateProfile({ profileImage: resp.success }).then((resp) => {
+            if (resp.profile) {
+              alert('Changed Succesfully');
+              console.log(resp);
```
Comment (rajivtitus) [inline client/src/pages/Profile/Profile.tsx]: Please remove console.log's throughout

File: server/controllers/profile.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+exports.uploadProfile = asyncHandler(async (req, res, next) => {
+    const userID = req.user.id;
+    const { profileImage, status } = req.body;
+    console.log(`profileImage: ${profileImage}`)
+    try {
+        let profile = await Profile.findOneAndUpdate({userID: userID},{profileImage: profileImage, status: status}, {new: true})
+        if (!profile) {            
+            profile = new Profile({userID: userID});
+            if (profileImage) {
+                console.log("Setting Image");
```
Comment (rajivtitus) [inline server/controllers/profile.js]: Please remove console.log's throughout

**Review by rajivtitus — APPROVED** [review]
LGTM! Please review comments and cleanup before merging. Thanks!

**Comment by dhanzy** [comment]
@rajivtitus . Okay I will work on the following comments.

File: client/src/interface/User.ts (hunk 2 above)
Comment (dhanzy) [reply client/src/interface/User.ts:7]: Concerning the '_id' i did not add that. Probably Hassan did.

Comment (dhanzy) [reply client/src/interface/User.ts:7]: @rajivtitus

**Comment by dhanzy** [comment]
Made neccessary comment changes to Profile Page @rajivtitus

File: client/src/interface/User.ts (hunk 2 above)
Comment (HAZberi) [reply client/src/interface/User.ts:7]: I was unable to access profile of because the controllers are set up as such.

**Pull request merged**